internal/controller/vrg_kubeobjects.go

```diff
@@ -154,34 +148,34 @@ func (v *VRGInstance) kubeObjectsCaptureStartOrResumeOrDelay(
 	)
 }
 
-func kubeObjectsCaptureStartConditionallySecondary(
+func kubeObjectsCaptureStartConditionally(
 	v *VRGInstance, result *ctrl.Result,
 	captureStartGeneration int64, captureStartTimeSince, captureStartInterval time.Duration,
 	captureStart func(),
 ) {
 	generation := v.instance.Generation
 	log := v.log.WithValues("generation", generation)
 
-	if captureStartGeneration == generation {
-		log.Info("Kube objects capture for relocate complete")
+	// Capture one last time for secondary before relocating
+	if v.instance.Spec.ReplicationState == ramen.Secondary &&
+		v.instance.Spec.Action != ramen.VRGActionRelocate {
```

Comment:
Where this condition came from? it was not in the previous code. This looks like the same issue as the previous PR - changing the behavior during refactoring.